core/AssetManager/UIAssetCacheBuster.php

```diff
@@ -33,14 +33,17 @@ public function piwikVersionBasedCacheBuster($pluginNames = false)
 
             $masterFile     = PIWIK_INCLUDE_PATH . '/.git/refs/heads/master';
             $currentGitHash = file_exists($masterFile) ? @file_get_contents($masterFile) : null;
+            $manager = Manager::getInstance();
 
-            $plugins = !$pluginNames ? Manager::getInstance()->getLoadedPluginsName() : $pluginNames;
+            $plugins = !$pluginNames ? $manager->getActivatedPlugins() : $pluginNames;
             sort($plugins);
 
             $pluginsInfo = '';
             foreach ($plugins as $pluginName) {
-                $plugin      = Manager::getInstance()->getLoadedPlugin($pluginName);
-                $pluginsInfo .= $plugin->getPluginName() . $plugin->getVersion() . ',';
+                if ($manager->isPluginLoaded($pluginName)) {
```

Comment:
Should we do this check here? If a plugin is in the activated plugins list, but for some reason isn't loaded, shouldn't the cache buster stay the same? I think we'd only change it if a plugin is activated or deactivated, not if a plugin isn't loaded for some reason.

Reply:
@diosmosis I'm not 100% sure but here's my thought. The asset manager posts an event to get all JS or CSS files. The event dispatcher in https://github.com/matomo-org/matomo/blob/4.3.1/core/EventDispatcher.php#L97 uses `getPluginsLoadedAndActivated()` and events will be only posted to plugins that are active and loaded. 

Meaning the hash would need to change if suddenly a plugin is loaded that wasn't loaded before because the JS/CSS would change too. I'm assuming a plugin may be activated but not loaded if it's missing from the filesystem (eg temporarily). Maybe there are also other reasons like when we unload a plugin in WhiteLabel etc.

I would probably keep the check but not 100% sure if it the thinking makes sense

Reply:
I see, no your logic makes sense. The plugins used are the ones that are specifically used to generate the JS/CSS, and there can be a discrepancy with the list of activated plugins and list of loaded plugins.